### PR TITLE
Refactor tests script

### PR DESCRIPTION
**Pull request by daschult**
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Changelog
*
*
*
*
*

### Description of the change

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@daschult__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (daschult). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by singhkays** [comment]
@daschult Can you summarize if there are new features to the CI in this PR or just refactoring?

**Comment by daschult** [comment]
I've updated the description of this change in the pull request comment.

**Comment by singhkays** [comment]
@daschult I'm missing where you added this. The description of the PR is boilerplate.

**Comment by daschult** [comment]
Oh, that's frustrating. I must've forgotten to save my edit.
This pull request is mostly just refactoring test/tests.js.
It does also add localTest.cmd, however, which allows test/tests.js to be run on a Windows computer. Users can specify whether they want remote validation and/or deployment to occur. Local validation will always occur, though.

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (singhkays) [inline .vscode/launch.json]: this folder seems .vscode specific. did you mean to include this?

**Comment by singhkays** [comment]
@simongdavies Since you've looked at this code before, could you give this a review as well?

File: .vscode/launch.json (hunk 1 above)
Comment (daschult) [inline .vscode/launch.json]: I did include this on purpose, because it's really useful when you're trying to debug the tests in VS Code. However, since it's tool specific, I'll remove it.

File: localTest.cmd [hunk 2]
```diff
@@ -0,0 +1,10 @@
+setlocal
+REM set VALIDATE_MODIFIED_ONLY=false
```
Comment (daschult) [inline localTest.cmd]: Yes, I wanted it to be a switch that could be turned on and off.

Comment (philon-msft) [inline localTest.cmd]: I'd add a comment to that effect

File: localTest.cmd [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+set PARALLEL_DEPLOYMENT_NUMBER=6
+set VALIDATION_HOST=http://qst9494f4f5c44dd94d84049.westus.cloudapp.azure.com
```
Comment (daschult) [inline localTest.cmd]: At this time I'll pass on this, but I'll consider it in the future.
